core/src/schema/namespace.rs

```diff
@@ -176,3 +242,109 @@ impl Compile for Namespace {
+                Content::SameAs(crate::schema::SameAsContent { ref_: ref2 }),
+            )
+            .unwrap();
+        namespace.put_collection(&"visitors".parse::<Name>().unwrap(), Content::Bool(BoolContent::Constant(true))).unwrap();
+        namespace.put_collection(&"daughters".parse::<Name>().unwrap(),Content::Bool(BoolContent::Constant(false) )).unwrap();
+        println!("sorted: {:?}", namespace.topo_sort());
+        let sorted = namespace.topo_sort().unwrap();
+        let length = sorted.len();
+        for i in 0..length {
+            assert!(check_dep(&sorted[..i], &namespace))
+        }
+    }
+
+    #[test]
+    fn test_sort_complex() {
+        let mut namespace = Namespace {
+            collections: BTreeMap::new(),
+        };
+        let ref1: FieldRef = "visitors.address.postcode".parse().unwrap();
+
+        namespace
+            .put_collection(
+                &"users".parse::<Name>().unwrap(),
+                Content::SameAs(crate::schema::SameAsContent { ref_: ref1.clone() }),
+            )
+            .unwrap();
+        namespace
+            .put_collection(
+                &"sons".parse::<Name>().unwrap(),
+                Content::SameAs(crate::schema::SameAsContent { ref_: ref1 }),
+            )
+            .unwrap();
+        println!("sorted: {:?}", namespace.topo_sort());
+        namespace.put_collection(&"visitors".parse::<Name>().unwrap(), Content::Bool(BoolContent::Constant(true))).unwrap();
+        namespace.put_collection(&"daughters".parse::<Name>().unwrap(),Content::Bool(BoolContent::Constant(false) )).unwrap();
+        
+        println!("sorted: {:?}", namespace.topo_sort());
+        let sorted = namespace.topo_sort().unwrap();
+        let length = sorted.len();
+        for i in 0..length {
+            assert!(check_dep(&sorted[..i], &namespace))
+        }
+    }
+
+    #[test]
+    fn test_sort_cycle() {
+        let mut namespace = Namespace {
+            collections: BTreeMap::new(),
+        };
+        let ref1: FieldRef = "visitors.address.postcode".parse().unwrap();
+        let ref2: FieldRef = "users.address.postcode".parse().unwrap();
+        //            let ref3: FieldRef = "winners.address.postcode".parse().unwrap();
+
+        namespace
+            .put_collection(
+                &"users".parse::<Name>().unwrap(),
+                Content::SameAs(crate::schema::SameAsContent { ref_: ref1 }),
+            )
+            .unwrap();
+        namespace
+            .put_collection(
+                &"visitors".parse::<Name>().unwrap(),
+                Content::SameAs(crate::schema::SameAsContent { ref_: ref2 }),
+            )
+            .unwrap();
+        println!("sorted: {:?}", namespace.topo_sort());
+        assert!(namespace.topo_sort().is_none());
+    }
+
+    //helper method for checking sorted dependencies
+    fn check_dep(list: &[Name], ns: &Namespace) -> bool {
+        let curr_name= match list.last() {
```

Comment:
How about `let (list, last) = list.split_at(list.len() - 1)`?

Reply:
Yes. This. Thanks

Reply:
Or `list.split_last()`?